docs/writing-tests/reftests.md

```diff
@@ -116,19 +116,19 @@ the dimensions of the image in pixels. Then <code>maxDifference =
 max<sub>x=[0,width) y=[0,height), c={r,g,b}</sub>(|T<sub>x,y,c</sub> -
 R<sub>x,y,c</sub>|)</code>.
 
-To specify the fuzziness in the test file one may add a `<meta
-name=fuzzy>` element (or, in the case of more complex tests, to any
-page containing the `<link rel=[mis]match>` elements). In the simplest
-case this has a `content` attribute containing the parameters above,
-separated by a colon e.g.
+To specify the fuzziness for a given test, one may add a `<meta name=fuzzy>`
+element to the test file (or, in the case of more complex tests that e.g. have
+subframes, to any page containing the `<link rel=[mis]match>` elements). The
+meta element should have a `content` attribute containing the parameters above,
+separated by a semicolon:
 
 ```
 <meta name=fuzzy content="maxDifference=15;totalPixels=300">
 ```
 
-would allow for a  difference of exactly 15 / 255 on any color channel
-and 300 exactly pixels total difference. The argument names are optional
-and may be elided; the above is the same as:
+would allow for a difference of 0-15 / 255 on any color channel and 0-300
```

Comment:
I think this was correct before; see https://searchfox.org/mozilla-central/source/testing/web-platform/tests/tools/manifest/sourcefile.py#611-615

Reply:
Why is it _max_ difference then? :(

This non-range option seems… confusing. Is it worthwhile changing this?

Reply:
It's maxDifference because you can have a difference of say 4 in one pixel and 15 in another and that will be fine as long as there's at least one that's exactly 15.

dbaron was previously keen to make it exact values rather than ranges on the basis that with ranges it's possible that the underlying issue causing fuzziness is resolved so all the differences go to 0, but no one notices because that's allowed by the annotation. Then you could miss a regression when the change is within the scope of the - now misleading - metadata.

Reply:
I'm now looking even more deeply at this, because we want to support fuzzy reftests in Chromium's CI today (long term, wptrunner, today we need to deal with blinkpy).  I was definitely wrong about the non-range option, it is equivalent to specifying a range of only that value (so `15` == `15-15`), and I agree with gsnedders that it seems very confusing, for both parameters.

I did a sweep of tests using fuzzy matching today. A random sampling of a few makes me suspect the syntax today is confusing test authors. All following were checked on wpt.fyi, then run via `wpt run` on my Linux desktop with a 1080p monitor:

* `css/css-images/image-orientation/image-orientation-default.html` - fails on all platforms, specifies 10;100 but produces 1;21 on my machine.
* `css/css-images/image-orientation/image-orientation-from-image-dynamic1.html` - fails on all platforms; specifies 10;100 but produces 1;21 on my machine.
* `css/css-transforms/perspective-transforms-equivalence.html` - fails on all platforms; specifies 10;10 but produces 0;0 on my machine.
* `html/rendering/non-replaced-elements/the-page/iframe-scrolling-attribute-values.html` - passes on all platforms; specifies 0-1;0-4, produces 0;0 on my machine.
* `density-size-correction/density-corrected-image-svg-aspect-ratio.html` - fails on all platforms; specifies 10;100 but produces 143;31014 on my machine (yeah this just seems borked).

So if the cases that are working are the ones providing a 0-N range, and the cases that are (incorrectly) failing are the ones providing an exact value... seems like we should either require it always be a range, or interpret `N` as `0-N`?

(Also, the CSS meta tag is maxDiff then totalPixels, but the logging output from wptrunner is totalPixels then maxDiff, grr...)

Side-note: the fact that the fuzziness is based on pixel also breaks all the tests on high-DPI devices too, right? That's unfortunate, though I know it is difficult to use e.g. percentages and not accidentally allow very bad false-matches.